models/src/main/scala/coop/rchain/models/rholangn/parmanager/RhoHash.scala

```diff
@@ -73,6 +73,7 @@ object RhoHash {
           /* Simple types */
           | _: ConnBoolN.type | _: ConnIntN.type | _: ConnBigIntN.type | _: ConnStringN.type |
           _: ConnUriN.type | _: ConnByteArrayN.type =>
+        // Terminals use serialized value as a source for hashing (AST base case)
```

Comment:
So we are forced to hash even terminals instead of just using serialised bytes?

Reply:
We need to prevent collisions. Without hashing e.g. string can be created the same as any hash.